src/schemas/zodSchema/signUpSchema.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+const SignUpSchema = z
+	.object({
+		firstName: z
+			.string()
+			.min(1, 'Please enter your firstname')
+			.regex(/^\w+$/, 'Firstname can only contain letters, numbers and/or underscore (_)'),
+
+		lastName: z
+			.string()
+			.min(1, 'Please enter your lastname')
+			.regex(/^\w+$/, 'Lastname can only contain letters, numbers and/or underscore (_)'),
+
+		email: z.string().email('Please enter a valid email address!'),
+
+		password: z
+			.string()
+			.min(12, 'Password must have at least 12 characters!')
+			.regex(/^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*\W).*$/, {
+				message: `Password must contain at least one uppercase letter, one lowercase letter,
+		  one number and one special character or symbol`,
+			}),
+
+		phoneNumber: z
+			.string()
+			.min(5, 'Phone number cannot be less than 5 digits!')
+			.max(20, 'Phone number cannot be more than 20 digits!')
+			.regex(/(?:([+]\d{1,4})[-.\s]?)?(?:[(](\d{1,3})[)][-.\s]?)?(\d{1,4})[-.\s]?(\d{1,4})[-.\s]?(\d{1,9})/g, {
+				message: 'Please enter a valid phone number!',
+			}),
+
+		gender: z.enum(['male', 'female', 'other', 'none']),
```

Comment:
There is a Gender enum provided already, why not use it to stay consistent when gender is updated.

Reply:
True, must've missed that

Reply:
But it seems doing just that like this:
gender: z.enum(Object.values(Gender))

Leads to ts errors like this:

Argument of type 'Gender[]' is not assignable to parameter of type '[string, ...string[]]'.

Reply:
But it seems doing just that like this:
`gender: z.enum(Object.values(Gender))`

Leads to ts errors like this:
`Argument of type 'Gender[]' is not assignable to parameter of type '[string, ...string[]]'.`

@Chimise

Reply:
Nevermind, just discovered a workaround for that

Reply:
@Chimise Nope, doesn't work still